tests/functional/monitoring/prometheus/alerts/test_alert_mds_cache_high_usage.py

```diff
@@ -0,0 +1,234 @@
+    2. Create dc pod with Fedora image
+    3. Copy helper_scripts/meta_data_io.py to Fedora dc pod
+    4. Run meta_data_io.py on fedora pod
+    """
+    access_mode = constants.ACCESS_MODE_RWX
+    file = constants.METAIO
+    interface = constants.CEPHFILESYSTEM
+    log.info("Checking for Ceph Health OK")
+    ceph_health_check_base()
+
+    for i in range(3):
+        # Creating PVC with cephfs as interface
+        log.info(f"Creating {interface} based PVC")
+        pvc_obj = pvc_factory(interface=interface, access_mode=access_mode, size="30")
+        # Creating a Fedora dc pod
+        log.info("Creating fedora dc pod")
+        pod_obj = dc_pod_factory(
+            pvc=pvc_obj, access_mode=access_mode, interface=interface
+        )
+        # Copy meta_data_io.py to fedora pod
+        log.info("Copying meta_data_io.py to fedora pod ")
+        cmd = f"oc cp {file} {pvc_obj.namespace}/{pod_obj.name}:/"
+        helpers.run_cmd(cmd=cmd)
+        log.info("meta_data_io.py copied successfully ")
+
+        # Run meta_data_io.py on fedora pod
+        log.info("Running meta data IO on fedora pod ")
+        metaio_executor = ThreadPoolExecutor(max_workers=1)
+        metaio_executor.submit(
+            pod_obj.exec_sh_cmd_on_pod, command="python3 meta_data_io.py"
+        )
+
+
+def active_mds_alert_values(threading_lock):
+    active_mds = cluster.get_active_mds_info()["mds_daemon"]
+    sr_mds = cluster.get_mds_standby_replay_info()["mds_daemon"]
+    cache_alert = constants.ALERT_MDSCACHEUSAGEHIGH
+    message = f"High MDS cache usage for the daemon mds.{active_mds}."
+    description = (
+        f"MDS cache usage for the daemon mds.{active_mds} has exceeded above 95% of the requested value."
+        f" Increase the memory request for mds.{active_mds} pod."
+    )
+    runbook = (
+        "https://github.com/openshift/runbooks/blob/master/alerts/"
+        "openshift-container-storage-operator/CephMdsCacheUsageHigh.md"
+    )
+    state = "firing"
+    severity = "critical"
+
+    api = prometheus.PrometheusAPI(threading_lock=threading_lock)
+    alerts_response = api.get("alerts", payload={"silenced": False, "inhibited": False})
+    prometheus_alerts = alerts_response.json()["data"]["alerts"]
+
+    prometheus.verify_mds_alerts(
+        alert_name=cache_alert,
+        msg=message,
+        description=description,
+        runbook=runbook,
+        state=state,
+        severity=severity,
+        alerts=prometheus_alerts,
+        active_mds=active_mds,
+        standby_mds=sr_mds,
+    )
+    log.info("Alert verified successfully")
+    return True
+
+
+@tier2
+@blue_squad
+class TestMdsMemoryAlerts:
+    alert_timer = 900  # sleep time to generate the alert 15 minutes
+    scale_timer = 100  # sleep time to wait after running scale down
+
+    @pytest.mark.polarion_id("OCS-5570")
+    def test_alert_triggered(
+        self, run_metadata_io_with_cephfs, threading_lock, alert_timer
+    ):
```

Comment:
Please add a description about the test case

Reply:
done